chat-sdk-ui/src/main/java/co/chatsdk/ui/ImageListAdapter.java

```diff
@@ -0,0 +1,107 @@
+package co.chatsdk.ui;
+
+import android.net.Uri;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+
+public class ImageListAdapter extends RecyclerView.Adapter<ImageListAdapter.MyViewHolder>  {
+
+    public ArrayList<Uri> uriArrayList = new ArrayList<>();
+    private WeakReference<SendMultipleImagesAtOnceActivity> theMainActivity;
+
+    public ImageListAdapter (SendMultipleImagesAtOnceActivity activity) {
+        this.theMainActivity = new WeakReference<SendMultipleImagesAtOnceActivity>(activity);
+    }
+
+    public class MyViewHolder extends RecyclerView.ViewHolder {
+
+        private WeakReference<ImageListAdapter> adapter;
+        public ImageView imageView;
+        public Uri thisImageUri;
+        public int thisImagePosition;
+
+        public MyViewHolder(ImageListAdapter la, ImageView v) {
+            super(v);
+            this.adapter = new WeakReference<ImageListAdapter>(la);
+            imageView = v;
+
+            imageView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    setMainImageDisplay(thisImageUri);
+                    setMainImageUri(thisImageUri);
+                }
+            });
+
+            imageView.setOnLongClickListener(new View.OnLongClickListener() {
+                @Override
+                public boolean onLongClick(View v) {
+                    Uri toBeDeletedUri = thisImageUri;
+                    removeImageFromAdapter(thisImageUri);
+                    notifyDataSetChanged();
+                    if (getMainImageUri() == toBeDeletedUri) {
+                        if (uriArrayList.size() >= 1 && uriArrayList.size() == thisImagePosition) {
+                            setMainImageDisplay(uriArrayList.get(thisImagePosition - 1));
+                            setMainImageUri(uriArrayList.get(thisImagePosition - 1));
+                        }
+                        else if (uriArrayList.size() > 1) {
+                            setMainImageDisplay(uriArrayList.get(thisImagePosition));
+                            setMainImageUri(uriArrayList.get(thisImagePosition));
+                        }
```

Comment:
I don't really understand this logic at all. Can you explain it?

Reply:
I changed it to just else. Basically, if you delete an image that was in the main view, then the imagePosition that that image was in now has a new image in it because the notifyOnDataSetChanged() method was called after the image was deleted. So calling setMainImageDisplay(imageURIs.get(imagePosition));
                            setMainImageUri(imageURIs.get(imagePosition));
causes a new image to be presented as the main image.